### PR TITLE
Support up-to-date `botocore`

### LINKED ISSUE #1056: Support botocore 1.32.1 with its massive decrease in package size
**Describe the bug**

The 1.32.1 release of botocore comes with a massive decrease (like 25% of the size it used to be, ~60MB smaller) in installed package size, which is great for decreasing the sizes of packages (e.g. lambdas): https://github.com/boto/botocore/issues/2365#issuecomment-1813447304

```python
pip install --no-deps botocore==1.32.1 --target /tmp/botocore-1.32.1
pip install --no-deps botocore==1.31.85 --target /tmp/botocore-1.31.85

du -sh /tmp/botocore-*
#  83M	/tmp/botocore-1.31.85
#  20M	/tmp/botocore-1.32.1
```

Currently aiobotocore supports up to botocore==1.31.64 with support for 1.31.70 pending in #1048.

The changes from #1048 to 1.32.1 seem to touch quite a few Python files in addition to the service JSON bumps. https://github.com/boto/botocore/compare/1.31.70...1.32.1

**Checklist**
N/A

**pip freeze results**
N/A

**Environment:**
N/A

**Additional context**

Thanks for aiobotocore! 

I get the impression that botocore upgrades are generally done "whenever", but this particular one seems so impactful that I'd like to have a place to track (this issue) so I can get a notification if/when we can upgrade. Thanks!

### PR DESCRIPTION
**Pull request by jakob-keller**
### Description of Change
This PR intends to improve general compatibility of `aiobotocore` within the Python ecosystem by bumping the dependency specification of `botocore`, as well as `boto3` and `awscli`.

### Assumptions
[Upstream diff](https://github.com/boto/botocore/compare/1.31.70..1.33.1) contains several minor changes that affect this codebase.

### Checklist for All Submissions
* [x] I have added change info to [CHANGES.rst](https://github.com/aio-libs/aiobotocore/blob/master/CHANGES.rst)
* [x] If this is resolving an issue (needed so future developers can determine if change is still necessary and under what conditions) (can be provided via link to issue with these details): resolves #1056
  * [ ] Detailed description of issue
  * [ ] Alternative methods considered (if any)
  * [x] How issue is being resolved: The required updates to `args.py`, `client.py`, and `utils.py` turned out to be trivial.
  * [ ] How issue can be reproduced
* [ ] If this is providing a new feature  (can be provided via link to issue with these details):
  * [ ] Detailed description of new feature
  * [ ] Why needed
  * [ ] Alternatives methods considered (if any)

### Checklist when updating botocore and/or aiohttp versions

* [x] I have read and followed [CONTRIBUTING.rst](https://github.com/aio-libs/aiobotocore/blob/master/CONTRIBUTING.rst#how-to-upgrade-botocore)
* [x] I have updated test_patches.py where/if appropriate (also check if no changes necessary)
* [x] I have ensured that the awscli/boto3 versions match the updated botocore version

### CODE REVIEW

**Comment by jakob-keller** [comment]
@thehesiod: This PR adds support for a whole bunch of new and updated AWS services that are [released during re:Invent 2023](https://aws.amazon.com/blogs/aws/top-announcements-of-aws-reinvent-2023/). I suggest to do a release soon to maintain feature parity.

`botocore=>1.33.2` includes many new codepaths and will likely be a major piece of work for us. This will need to be worked on in another PR.

**Comment by thehesiod** [comment]
that's so awesome you added link to diff.  We need to add that to the template in the checklist with example

**Comment by thehesiod** [comment]
looks great, thanks!

File: setup.py [hunk 1]
```diff
@@ -7,15 +7,15 @@
 # NOTE: When updating botocore make sure to update awscli/boto3 versions below
 install_requires = [
     # pegged to also match items in `extras_require`
-    'botocore>=1.31.16,<1.31.71',
+    'botocore>=1.32.4,<1.33.2',
```
Comment (thehesiod) [inline setup.py:10]: I missed this in the review, it's doubtful this aiobotocore is compatible across these versions given it required changes to match the latest botocore

**Pull request merged**